ui/components/multichain/create-named-snap-account/create-named-snap-account.test.tsx

```diff
@@ -0,0 +1,181 @@
+      completedOnboarding: true,
+      internalAccounts: {
+        ...mockState.metamask.internalAccounts,
+        accounts: {
+          ...mockState.metamask.internalAccounts.accounts,
+          [mockSnapAccount1.id]: mockSnapAccount1,
+          [mockSnapAccount2.id]: mockSnapAccount2,
+        },
+        options: {},
+        methods: ETH_EOA_METHODS,
+        type: 'eip155:eoa',
+      },
+    },
+  });
+  return renderWithProvider(<CreateNamedSnapAccount {...props} />, store);
+};
+
+describe('CreateNamedSnapAccount', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('displays account name input and suggested name', async () => {
+    const { getByPlaceholderText } = render();
+
+    await waitFor(() =>
+      expect(
+        getByPlaceholderText(mockSnapSuggestedAccountName),
+      ).toBeInTheDocument(),
+    );
+  });
+
+  it('renames account and fires onActionComplete with true when clicking "Add account"', async () => {
+    const onActionComplete = jest.fn();
+    const { getByText, getByPlaceholderText } = render({
+      onActionComplete,
+      address: mockAccount.address,
+      snapSuggestedAccountName: mockSnapSuggestedAccountName,
+    });
+
+    const input = await waitFor(() =>
+      getByPlaceholderText(mockSnapSuggestedAccountName),
+    );
+    const newAccountName = 'New Account Name';
+
+    fireEvent.change(input, {
+      target: { value: newAccountName },
+    });
+    fireEvent.click(getByText('Add account'));
+
+    await waitFor(() =>
+      expect(onActionComplete).toHaveBeenCalledWith({
+        success: true,
+        name: newAccountName,
+      }),
+    );
+  });
+
+  it(`doesn't allow duplicate account names`, async () => {
+    const { getByText, getByPlaceholderText } = render();
+
+    const input = await waitFor(() =>
+      getByPlaceholderText(mockSnapSuggestedAccountName),
+    );
+    const usedAccountName = 'Snap Account 1';
+
+    fireEvent.change(input, {
+      target: { value: usedAccountName },
+    });
+
+    const submitButton = getByText('Add account');
+    expect(submitButton).toHaveAttribute('disabled');
+  });
+
+  it('uses default account name when input is empty and fires onActionComplete with true when clicking "Add account"', async () => {
+    // Note: last account index for snap keyring
+    const defaultAccountName = 'Snap Account 2';
+
+    const onActionComplete = jest.fn();
+    const { getByText, getByPlaceholderText } = render({
+      onActionComplete,
+      address: mockAccount.address,
+    });
+
+    fireEvent.click(getByText('Add account'));
+
+    await waitFor(() => {
+      // Check if the input value has been updated to the default account name
+      expect(getByPlaceholderText(defaultAccountName)).toBeInTheDocument();
+      expect(onActionComplete).toHaveBeenCalledWith({
+        success: true,
+        name: '',
+      });
+    });
+  });
+
+  it('fires onActionComplete with false when clicking Cancel', async () => {
```

Comment:
Cosmetic to match the previous test:

```suggestion
  it('fires onActionComplete with false when clicking "Cancel"', async () => {
```